pkg/syncer/cnsoperator/controller/cnsnodevmbatchattachment/cnsnodevmbatchattachment_helper.go

```diff
@@ -470,3 +504,245 @@ func getVolumesToDetachFromVM(ctx context.Context, client client.Client,
+
+	if pvc.Annotations == nil {
+		log.Infof("PVC %s does not have any annotations, nothing to be done", pvc.Name)
+		return nil
+	}
+
+	return patchPVCAnnotations(ctx, k8sClient, pvc, attachedVmPrefix+vmInstanceUUID, true)
+}
+
+// patchPVCAnnotations patches the list of annotations on the PVC with the newAnnotations.
+func patchPVCAnnotations(ctx context.Context, k8sClient kubernetes.Interface,
+	pvc *v1.PersistentVolumeClaim, key string, removeAnnotation bool) error {
+	log := logger.GetLogger(ctx)
+
+	patchAnnotations := make(map[string]interface{})
+	for k, v := range pvc.Annotations {
+		patchAnnotations[k] = v
+	}
+
+	if removeAnnotation {
+		log.Infof("Removing annotation %s from PVC %s", key, pvc.Name)
+		patchAnnotations[key] = nil
+	} else {
+		log.Infof("Adding annotation %s on PVC", key, pvc.Name)
+		patchAnnotations[key] = ""
+	}
+
+	log.Debugf("Updated list of annotations on PVC %s: %+v", pvc.Name, patchAnnotations)
+
+	// Build patch structure
+	patch := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"annotations": patchAnnotations,
+		},
+	}
+
+	patchBytes, err := json.Marshal(patch)
+	if err != nil {
+		log.Errorf("failed to marshal with annotations for PVC %s. Err: %s", pvc.Name, err)
+		return fmt.Errorf("failed to marshal patch: %v", err)
+	}
+
+	log.Infof("Patching PVC %s with updated annotation", pvc.Name)
+
+	// Apply the patch
+	updatedpvc, err := k8sClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Patch(
+		ctx,
+		pvc.Name,
+		types.MergePatchType,
+		patchBytes,
+		metav1.PatchOptions{},
+	)
+	if err != nil {
+		log.Errorf("failed to patch PVC %s with annotations. Err: %s", pvc.Name, err)
+		return fmt.Errorf("failed to patch PVC %s: %v", pvc.Name, err)
+	}
+	log.Debugf("Successfully patched PVC: %s with annotations %+v", pvc.Name, updatedpvc.Annotations)
+	return nil
+}
+
+// pvcHasUsedByAnnotaion goes through all annotations on the PVC to find out if the PVC is used by any VM or not.
+func pvcHasUsedByAnnotaion(ctx context.Context, pvc *v1.PersistentVolumeClaim) bool {
+	log := logger.GetLogger(ctx)
+
+	if pvc.Annotations == nil {
+		log.Infof("No annotation found on PVC %s", pvc.Name)
+		return false
+	}
+
+	for key := range pvc.Annotations {
+		if strings.HasPrefix(key, attachedVmPrefix) {
+			log.Infof("Annotation with prefix %s found on PVC %s", attachedVmPrefix, pvc.Name)
+			return true
+		}
+	}
+
+	log.Infof("PVC %s does not contain any annotations with prefix %s", pvc.Name, attachedVmPrefix)
+	return false
+}
+
+// addPvcFinalizer adds the given VM as an annotation for the given PVC and
+// adds CNS finalizer to the PVC.
+func addPvcFinalizer(ctx context.Context, client client.Client,
+	k8sClient kubernetes.Interface,
+	pvcName string, namespace string, vmInstanceUUID string) error {
+	log := logger.GetLogger(ctx)
+
+	// Acquire lock on PVC
+	namespacedVolumeName := namespace + "/" + pvcName
+	actual, _ := VolumeLock.LoadOrStore(namespacedVolumeName, &sync.Mutex{})
```

Comment:
can we give better name for variable "actual"

Reply:
func (m *Map) LoadAndStore(key, value any) (actual any, loaded bool)

This is the signature of LoadAndStore. It returns actual that is why we use it like that everywhere.